src/interpolation_fe.jl

```diff
@@ -0,0 +1,176 @@
+# In this tutorial, we will look at how to
+# - Evaluate `CellFields` at arbitrary points
+# - Interpolate finite element functions defined on different
+# triangulations. We will consider examples for
+#    - Lagrangian finite element spaces
+#    - Vector-Valued Spaces
+#    - Raviart Thomas finite element spaces
+#    - Multifield finite element spaces
+
+# ## Problem Statement
+# Let $\mathcal{T}_1$ and $\mathcal{T}_2$ be two triangulations of a
+# domain $\Omega$. Let $V_i$ be the finite element space defined on
+# the triangulation $\mathcal{T}_i$ for $i=1,2$. Let $f_h \in V_1$. The
+# interpolation problem is to find $g_h \in V_2$ such that
+#
+# ```math
+# dof_k^{V_2}(g_h) = dof_k^{V_2}(f_h),\quad \forall k \in
+# \{1,\dots,N_{dof}^{V_2}\}
+# ```
+
+
+# ## Setup
+# For the purpose of this tutorial we require `Test`, `Gridap` along with the
+# following submodules of `Gridap`
+
+using Test
+using Gridap
+using Gridap.CellData
+using Gridap.Visualization
+
+# We now create a computational domain on the unit square $[0,1]^2$ consisting
+# of 5 cells per direction
+
+domain = (0,1,0,1)
+partition = (5,5)
+𝒯₁ = CartesianDiscreteModel(domain, partition)
+
+
+# ## Background
+# `Gridap` offers the feature to evaluate functions at arbitrary
+# points in the domain. This will be shown in the next
+# section. Interpolation then takes advantage of this feature to
+# obtain the `FEFunction` in the new space from the old one by
+# evaluating the appropriate degrees of freedom. Interpolation works
+# using the composite type `Interpolable` to tell `Gridap` that the
+# argument can be interpolated between triangulations. This can be
+# done using any of the methods
+# ```julia
```

Comment:
I think this risks not ageing well... because, if for some reason we change the name of `interpolate` or `interpolate_dirichlet`, it will be hard to find out that this comment is outdated. I would remove this comment block and below, when you first call `interpolate_everywhere`, point out that you can also do `interpolate` or `interpolate_dirichlet` and execute both in a cell. In this way, if these procedures change, this tutorial will also fail.